pages/api/auth/forgot-password.ts

```diff
@@ -0,0 +1,77 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import prisma from "../../../lib/prisma";
+import dayjs from "dayjs";
+import { User, ResetPasswordRequest } from "@prisma/client";
+import sendEmail from "../../../lib/emails/sendMail";
+import { buildForgotPasswordMessage } from "../../../src/forgot-password/messaging/forgot-password";
+import timezone from "dayjs/plugin/timezone";
+import utc from "dayjs/plugin/utc";
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse) {
+  if (req.method !== "POST") {
+    return res.status(400).json({ message: "" });
+  }
+
+  try {
+    const rawEmail = req.body?.email;
+
+    const maybeUser: User = await prisma.user.findUnique({
+      where: {
+        email: rawEmail,
+      },
+      select: {
+        name: true,
+      },
+    });
+
+    if (!maybeUser) {
+      return res.status(400).json({ message: "Couldn't find an account for this email" });
+    }
+
+    const now = dayjs().toDate();
+    const maybePreviousRequest = await prisma.resetPasswordRequest.findMany({
+      where: {
+        email: rawEmail,
+        expires: {
+          gt: now,
+        },
+      },
+    });
+
+    let passwordRequest: ResetPasswordRequest;
+
+    if (maybePreviousRequest && maybePreviousRequest?.length >= 1) {
+      passwordRequest = maybePreviousRequest[0];
+    } else {
+      const expiry = dayjs().tz(maybeUser.timeZone).add(6, "hours").toDate();
```

Comment:
To be honest; looking at the database; it already stores the timestamp in UTC
```
calendso2=# select * from "ResetPasswordRequest";
-[ RECORD 1 ]------------------------------
id        | ckqbifqz100021nvhqd8anmf0
createdAt | 2021-06-24 23:01:58.477
updatedAt | 2021-06-24 23:01:58.478
email     | avanandel@...
expires   | 2021-06-25 05:01:58.363`
```
To simplify I suggest using the following; also means you can ditch the timeZone module from this page, double win.
```javascript
dayjs.utc().add(6 "hours").toDate()
// or even dayjs().add(6 "hours").toDate()
```
tz really does not matter here as later on you are comparing against server time

Reply:
This is good to know.